### PR TITLE
doc: Adding a new simple demo - first steps

### PR DESCRIPTION
**Pull request by draychev**
This PR adds a new demo, which attempts to be the simplest possible demo, which shows:
  1. how to download and install OSM
  2. create 2 pods
  3. apply an SMI policy to allow traffic between the 2 pods.

The goal is to create a series of demos, which build on each other, are very easy to follow, and walk us through everything the current manual demo does.

ref #2854

### CODE REVIEW

**Review by shashankram — COMMENTED** [review]
Should demos be housed in a separate repo?

File: docs/demos/first_steps.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+aliases: ["OSM Simple Demo"]
+weight: 2
+release: 0.8.0-rc.2
+---
+
+
+# Open Service Mesh Demo: First Steps
+
+
+> Note: This demo is specific to OSM v0.8.0-rc.2
```
[the remark this answers is not in the document]
Comment (draychev) [reply docs/demos/first_steps.md:14]: @nojnhuh I plan on making a dozen of these smaller demos.  After we created them all - we can add them to the release guide.  For that I created this issue:  https://github.com/openservicemesh/osm/issues/2908

**Comment by draychev** [comment]
> Should demos be housed in a separate repo?

The goal is to create broad and deep content covering all features in v0.8.  Once we have this content we should a) link between the various topics and b) reorganize.

Once we have everything written (from https://github.com/openservicemesh/osm/issues/2854) we can step back - look at it - and then make a decision on what the organization should look like.

I created this task to track this as a follow-up task:  https://github.com/openservicemesh/osm/issues/2909

**Review by nojnhuh — APPROVED** [review]

File: docs/demos/first_steps.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+## Installing OSM on Kubernetes
+
+With the `osm` binary downloaded and unzipped, we are ready to install Open Service Mesh on a Kubernetes cluster:
+
+```bash
+osm install
+```
+
+> Note: This document assumes you have already installed credentials for a Kubernetes cluster in ~/.kube/config and `kubectl cluster-info` executes successfully.
```
Comment (shashankram) [inline docs/demos/first_steps.md:67]: Does the CLI require the kubeconfig file to be present? This should be documented as a prerequisite to running `osm install` if that's the case.

File: docs/demos/first_steps.md [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+
+This installed OSM Controller in the `osm-system` namespace.
```
Comment (shashankram) [inline docs/demos/first_steps.md:69]: `installed` assumes the installation was successful. Does it make sense to reword it to `should install`, and add a note on how to check if the install was successful or not?

File: docs/demos/first_steps.md [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+## Deploy Demo Applications
+In this section we will deploy 2 pods in 2 different namespaces:
+- `bookbuyer` - a client, which makes HTTP requests to a `bookstore`
+- `bookstore` - a server, which responds to HTTP requests from `bookbuyer`
```
Comment (shashankram) [inline docs/demos/first_steps.md:74]:
```suggestion
- `bookbuyer` - an HTTP client, which makes HTTP requests to a `bookstore`
- `bookstore` - an HTTP server, which responds to HTTP requests from `bookbuyer`
```

File: docs/demos/first_steps.md [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+
+### Create the Namespaces
+Each application in this demo will reside in a separate namespaces. Create the namespaces with:
```
Comment (shashankram) [inline docs/demos/first_steps.md:78]:
```suggestion
Each application in this demo will reside in separate namespaces. Create the namespaces with:
```

File: docs/demos/first_steps.md [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+```
+
+### Join the new namespaces to the service mesh
+The following `osm` CLI command joins the namespaces to the service mesh:
+
+`//TODO(draychev): link to relevant documentation on how the JOIN command works [#2904]`
+
+```bash
+osm namespace add bookbuyer
+osm namespace add bookstore
```
Comment (shashankram) [inline docs/demos/first_steps.md:91]: Why not `osm namespace add bookbuyer bookstore` ?

File: docs/demos/first_steps.md [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+```
+
+With this command each one of the four namespaces will be:
```
Comment (shashankram) [inline docs/demos/first_steps.md:94]:
```suggestion
With this command each one of the two namespaces will be:
```

File: docs/demos/first_steps.md [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+Create the `bookbuyer` service account:
+```bash
+kubectl apply -f - <<EOF
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: bookbuyer
+  namespace: bookbuyer
+EOF
```
Comment (shashankram) [inline docs/demos/first_steps.md:112]: `<<EOF` is specific to unix shells, is it noted somewhere in the guide these commands are meant to be run in such an environment?

File: docs/demos/first_steps.md [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+  namespace: bookbuyer
+  name: bookbuyer
+  labels:
+    app: bookbuyer
+    version: v1
+spec:
+  serviceAccountName: bookbuyer
+  containers:
+  - name: bookbuyer
+    image: openservicemesh/bookbuyer:v0.8.0-rc.2
```
Comment (shashankram) [inline docs/demos/first_steps.md:131]: Maybe use the `latest` tag for the images so they always point to the image tags from the latest release?

File: docs/demos/first_steps.md [hunk 10]
```diff
@@ -0,0 +1,299 @@
...
+EOF
+```
+
+To view the resources we created so far run the following commands:
+
+```bash
+kubectl get pods -A | grep -E '^book'
+
+kubectl get services -A | grep -E '^book'
+```
```
Comment (shashankram) [inline docs/demos/first_steps.md:202]: Explicitly specify the namespace here so the commands are not specific to posix shells.

File: docs/demos/first_steps.md [hunk 11]
```diff
@@ -0,0 +1,299 @@
...
+
+We have not applied any traffic policies yet.
+[Service Mesh Interface](https://smi-spec.io/) (and OSM by proxy) denies requests where no explicit policy is set.
+To permit HTTP GET calls from `bookbuyer` to `bookstore` we need to apply an SMI policy.
+
+## Applying SMI Policies
+
+With SMI policies we are going to:
+
+1. authorize access between pods (service accounts)
```
Comment (shashankram) [inline docs/demos/first_steps.md:233]:
```suggestion
1. authorize access between pods (using their service accounts)
```

File: docs/demos/first_steps.md [hunk 12]
```diff
@@ -0,0 +1,299 @@
...
+2. define routing rules for granular access control
```
Comment (shashankram) [inline docs/demos/first_steps.md:234]:
```suggestion
2. define access control and routing rules for granular traffic management
```

**Review by shashankram — CHANGES_REQUESTED** [review]

File: docs/demos/first_steps.md (hunk 9 above)
Comment (draychev) [reply docs/demos/first_steps.md:131]: Our docs are going to be versioned. We are always going to reference a specific version of OSM.

File: docs/demos/first_steps.md (hunk 3 above)
Comment (draychev) [reply docs/demos/first_steps.md:69]: I'd like to keep this as minimal as possible. There are many error scenarios. This doc assumes everything goes smoothly. It is meant to be a quick breeze-through.
We will handle error scenarios in other areas of the documentation.

File: docs/demos/first_steps.md (hunk 6 above)
Comment (draychev) [reply docs/demos/first_steps.md:91]: I prefer having them on 2 separate lines for clarity - we create 2 namespaces - here are 2 lines. No ambiguity.

File: docs/demos/first_steps.md (hunk 8 above)
Comment (draychev) [reply docs/demos/first_steps.md:112]: One of the 3 requirements for this demo is `  - a workstation capable of executing [Bash](https://en.wikipedia.org/wiki/Bash_(Unix_shell)) scripts`

heredocs are extremely nice for docs like that

File: docs/demos/first_steps.md (hunk 2 above)
Comment (draychev) [reply docs/demos/first_steps.md:67]: The `~/.kube/config` file is by convention where the credentials for the kube client are. Without this file neither `kubectl` nor `osm` will work.
Example:
```
$ mv ~/.kube/config ~/.kube/config___
$ ./osm mesh list
Error: Could not list deployments Get "http://localhost:8080/apis/apps/v1/deployments?labelSelector=app%3Dosm-controller": dial tcp 127.0.0.1:8080: connect: connection refused


$ mv ~/.kube/config___ ~/.kube/config; osm mesh list

MESH NAME       NAMESPACE       CONTROLLER PODS
osm-many-many   osm-many-many   osm-controller-5b55f68568-8r46s
osm             osm-system      osm-controller-666f54645d-gqcn6
```

**Review by shashankram — APPROVED** [review]
